src/k8s-engine/wsl.ts

```diff
@@ -1304,6 +1302,22 @@ export default class WSLBackend extends events.EventEmitter implements K8s.Kuber
           this.progressTracker.action('Installing image scanner', 100, this.installTrivy()),
           this.progressTracker.action('Installing CA certificates', 100, this.installCACerts()),
           this.progressTracker.action('Installing helpers', 50, this.installWSLHelpers()),
+          this.progressTracker.action('Writing K3s configuration', 50, async() => {
+            const k3sConf = {
+              PORT:                   this.#desiredPort.toString(),
+              LOG_DIR:                await this.wslify(paths.logs),
+              'export IPTABLES_MODE': 'legacy',
+              ENGINE:                 this.#currentContainerEngine,
+              ADDITIONAL_ARGS:        this.cfg?.options.traefik ? '' : '--disable traefik',
+            };
+
+            if (!this.cfg?.options.flannel) {
+              console.log(`Disabling flannel and network policy`);
+              k3sConf.ADDITIONAL_ARGS += '--flannel-backend=none --disable-network-policy';
```

Comment:
If I set `options.flannel` to `false` I get this line in `/etc/conf.d/k3s`:
```
ADDITIONAL_ARGS="--disable traefik--flannel-backend=none --disable-network-policy"
                                                             ^
```
It was like that before, but needs fixing, or maybe this should go in a separate issue, but I'm not sure we can currently test it. Leaving it up to you.